### PR TITLE
3.0 media tags

### PR DESCRIPTION
**Pull request by falkodev**
https://apostrophecms.atlassian.net/browse/A30U-115

### CODE REVIEW

File: lib/modules/@apostrophecms/files-tags/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    fields: {
+      add: {
+        title: {
```
Comment (boutell) [inline lib/modules/@apostrophecms/files-tags/index.js:6]: This is correct, but is it necessary? There is already a title field configuration for all doc types by default.

File: lib/modules/@apostrophecms/files/index.js [hunk 2]
```diff
@@ -45,6 +45,12 @@ module.exports = {
         type: 'url',
         name: 'creditUrl',
         label: 'Credit URL'
+      },
+      {
+        type: 'joinByArray',
+        name: '_tags',
+        label: 'Tags',
+        withType: '@apostrophecms/files-tags'
```
Comment (boutell) [inline lib/modules/@apostrophecms/files/index.js:53]: 👍

File: lib/modules/@apostrophecms/files-tags/index.js (hunk 1 above)
Comment (falkodev) [reply lib/modules/@apostrophecms/files-tags/index.js:7]: Oh ok, I thought I had to add this field. Will remove it.

File: lib/modules/@apostrophecms/files-tags/index.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Files Tag'
```
Comment (falkodev) [inline lib/modules/@apostrophecms/files-tags/index.js:4]: Should it be `File Tag`? @boutell

Comment (boutell) [reply lib/modules/@apostrophecms/files-tags/index.js:4]: I think "File Tag" reads better, yes.

File: lib/modules/@apostrophecms/schemas/index.js [hunk 4]
```diff
@@ -769,12 +769,12 @@ module.exports = {
         if (Array.isArray(field.withType)) {
           _.each(field.withType, function (type) {
             if (!_.find(self.apos.docs.managers, { name: type })) {
-              fail('withType property, ' + type + ', does not match the "name" property of any doc type. Hint: this is not the same thing as a module name. Usually singular.');
+              fail('withType property, ' + type + ', does not match the "name" property of any doc type. You have to set it as the name of the joined module.');
```
[the remark this answers is not in the document]
Comment (boutell) [reply lib/modules/@apostrophecms/schemas/index.js:772]: How about: "in most cases this is the same as the module name."

**Pull request merged**